### ISSUE TITLE
Unable to invoke tenant template: {"SiteId":"","SiteStatus":3,"SiteUrl":""} Exception (Invoke-PnPTenantTemplate)

### ISSUE THREAD

**Issue by arnieraju** [pnp/pnpframework#232]
Hello,

I am trying to use the Invoke-PnPTenantTemplate to apply a template that I have just generated using Get-PnPTenantTemplate, however am receiving the exception below.  

```
Message          : {"SiteId":"","SiteStatus":3,"SiteUrl":""}
Stacktrace       :    at PnP.Framework.Provisioning.ObjectHandlers.SiteToTemplateConversion.ApplyTenantTemplate(Tenant tenant, ProvisioningHierarchy hierarchy, String sequenceId, ApplyConfiguration configuration)
                      at PnP.PowerShell.Commands.Provisioning.Tenant.InvokeTenantTemplate.ExecuteCmdlet()
                      at PnP.PowerShell.Commands.PnPSharePointCmdlet.ProcessRecord()
ScriptLineNumber : 1
```

![image](https://user-images.githubusercontent.com/1208184/112259569-4126dd00-8cb4-11eb-90bc-e3d3d22fc3df.png)

I am not sure whether this is user error (possibly) or whether I am hitting a bug.  The message isn't too helpful, but it did get me thinking that the parameters weren't being passed through.  Below are details about how I am running the command.  I have also tried executing it against two different tenancies with the same outcome for each. 

I am using the version 1.4.0 of PnP.PowerShell (PnP PowerShell Cmdlets (1.4.0.0)) and connecting as a Global Admin using 
```
connect-pnponline -Url https://foobar-admin.sharepoint.com -Interactive
```

I have generated the template using Get-PnPTenantTemplate successfully using the following configuration:
```
{
    "$schema":"https://aka.ms/sppnp-extract-configuration-schema",
    "handlers": [
      "WebSettings",
      "Pages",
      "ContentTypes",
      "Fields",
      "Lists"
    ],
    "persistAssetFiles": true,
    "pages": {"includeAllClientSidePages": true, "excludeAuthorInformation": true},
    "navigation": {"removeExistingNodes": true},
    "contentTypes": {"groups": ["Intranet"]},
    "lists": {"includeHiddenLists": false, "lists": [{"title": "Documents", "includeItems":true, "skipEmptyFields": true},{"title": "Site Pages", "includeItems":true, "skipEmptyFields": true}]},
    "tenant": { "sequence": {"siteUrls": ["https://foobar.sharepoint.com/sites/intranet-starter"], "includeJoinedSites": true}}
  }
```

I then initialised a $parameters object using the code below:

```
    $tenantSitesUrl = "https://foobar.sharepoint.com/sites";
    $intranetUrlPrefix = "intranet-starter";

    $intranetSiteUrl = "$tenantSitesUrl/$intranetUrlPrefix-$siteSuffix";
    $intranetSiteName = "Intranet Starter $siteSuffix"
    $intranetNewsSiteUrl = "$tenantSitesUrl/$intranetUrlPrefix-news-and-events-$siteSuffix";
    $intranetNewsSiteName = "Intranet News and Events $siteSuffix";
    $intranetInfoSiteUrl = "$tenantSitesUrl/$intranetUrlPrefix-info-$siteSuffix";
    $intranetInfoSiteName = "Intranet Information and Services $siteSuffix";

    $parameters = @{
        "SITECOLLECTION_9c39465ff78d4ddfb1c8a328f711a284_URL" = $intranetSiteUrl; # intranet site url
        "SITECOLLECTION_9c39465ff78d4ddfb1c8a328f711a284_TITLE" = $intranetSiteName; # intranet site name
        "SITECOLLECTION_c67c3d0faf0e4690b854cf20c4ea97c5_URL" = $intranetNewsSiteUrl; # intranet site url
        "SITECOLLECTION_c67c3d0faf0e4690b854cf20c4ea97c5_TITLE" = $intranetNewsSiteName; # intranet site name
        "SITECOLLECTION_8378f627313a49a6a91abd005bd9df41_URL" = $intranetInfoSiteUrl # intranet site url
        "SITECOLLECTION_8378f627313a49a6a91abd005bd9df41_TITLE" = $intranetInfoSiteName; # intranet site name
    };
```

Which comes out as
```
Name                           Value
----                           -----
SITECOLLECTION_c67c3d0faf0e... https://foobar.sharepoint.com/sites/intranet-starter-news-and-events-01
SITECOLLECTION_8378f627313a... https://foobar.sharepoint.com/sites/intranet-starter-info-01
SITECOLLECTION_9c39465ff78d... https://foobar.sharepoint.com/sites/intranet-starter-01
SITECOLLECTION_c67c3d0faf0e... Intranet News and Events 01
SITECOLLECTION_9c39465ff78d... Intranet Starter 01
SITECOLLECTION_8378f627313a... Intranet Information and Services 01
```

and aligns with the templates following parameters in the template xml:
```
    <pnp:Parameters>
      <pnp:Parameter Key="SITECOLLECTION_9c39465ff78d4ddfb1c8a328f711a284_URL">https://foobar.sharepoint.com/sites/intranet-starter</pnp:Parameter>
      <pnp:Parameter Key="SITECOLLECTION_9c39465ff78d4ddfb1c8a328f711a284_TITLE">School of XYZ intranet</pnp:Parameter>
      <pnp:Parameter Key="SITECOLLECTION_c67c3d0faf0e4690b854cf20c4ea97c5_URL">https://foobar.sharepoint.com/sites/intranet-starter-news-and-events</pnp:Parameter>
      <pnp:Parameter Key="SITECOLLECTION_c67c3d0faf0e4690b854cf20c4ea97c5_TITLE">News and events - School of XYZ</pnp:Parameter>
      <pnp:Parameter Key="SITECOLLECTION_8378f627313a49a6a91abd005bd9df41_URL">https://foobar.sharepoint.com/sites/intranet-starter-information-services-and-support</pnp:Parameter>
      <pnp:Parameter Key="SITECOLLECTION_8378f627313a49a6a91abd005bd9df41_TITLE">Information, services and support - School of XYZ</pnp:Parameter>
    </pnp:Parameters>

```

and then ran
```
 Invoke-PnPTenantTemplate .\StandardTemplate.xml -ResourceFolder .\SiteAssets\ -Parameters $parameters
```

This was all completed in the directory where StandardTemplate resides
```
Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----        3/24/2021  10:35 AM                Shared Documents
d-----        3/24/2021  10:50 AM                SiteAssets
d-----        3/24/2021  10:53 AM                SitePages
-a----        3/24/2021  10:56 AM        2373192 StandardTemplate.xml
```

Note: I haven't included the template here as it is quite large (lots of images and page content). I am about to create a 'hello world' template and see what results I get with that.  I will report back with results from that test.

**Comment by jansenbe** [comment]
@arnieraju : the error indicates that something went wrong with creating the site collection defined in the template. Can you cross check that you're not creating a site collection that was already created?

**Comment by arnieraju** [comment]
@jansenbe thanks so much for the quick reply.  Yes it bombed out pretty quickly, so I did think it must have prior to sites being created.   I just double checked and can confirm that sites collections do not already exist.  The URLs that are generated are unique.

I also ran the script again adding a new suffix (188), and had the same result.  In the progress bar at the top, you can see that it flashes up Creating site collection ... and then I get the same exception coming up.

```
Message          : {"SiteId":"","SiteStatus":3,"SiteUrl":""}
Stacktrace       :    at PnP.Framework.Provisioning.ObjectHandlers.SiteToTemplateConversion.ApplyTenantTemplate(Tenant
                   tenant, ProvisioningHierarchy hierarchy, String sequenceId, ApplyConfiguration configuration)
                      at PnP.PowerShell.Commands.Provisioning.Tenant.InvokeTenantTemplate.ExecuteCmdlet()
                      at PnP.PowerShell.Commands.PnPSharePointCmdlet.ProcessRecord()
ScriptLineNumber : 1
```

Parameters this time were
```
Name                           Value
----                           -----
SITECOLLECTION_c67c3d0faf0e... https://thesocodev1.sharepoint.com/sites/intranet-starter-news-and-events-188
SITECOLLECTION_8378f627313a... https://thesocodev1.sharepoint.com/sites/intranet-starter-info-188
SITECOLLECTION_9c39465ff78d... https://thesocodev1.sharepoint.com/sites/intranet-starter-188
SITECOLLECTION_c67c3d0faf0e... Intranet News and Events 188
SITECOLLECTION_9c39465ff78d... Intranet Starter 188
SITECOLLECTION_8378f627313a... Intranet Information and Services 188
```

**Comment by arnieraju** [comment]
A quick update from further testing on my side.  When running a simple hello-world test, creating a template from a comms site with a single page with an image on it, and then importing it it works without issue.

In the Tenant Template that I am experiencing issues with I removed everything in the XML (fields, pages, lists etc) except for what is needed to create of the main site (there are 3 sites, one of which is a hub).  With a bare XML file the Invoke-PnPTenantTemplate command does successfully manage to create the site collection.

So it looks like there may be an issue with the contents of the template and how that is handled by the provisioning API.  

In terms of next steps, @jansenbe do you or the team have any tips on how I can efficiently narrow down what may be leading to the exception?  I've got a few ideas but thought it might be worth getting a steer from you before I go down a rabbit hole :)

**Comment by arnieraju** [comment]
Hi @jansenbe and Team!  I'm about to start looking into this one again.  Based on my results so far is there any guidance you can provide on efficiently tracking down the issue?  Cheers Arnie

**Comment by jansenbe** [comment]
@arnieraju : I'm confused as your error message indicates that an issue with the site collection creation while you say in the last updates it's related to the contents of the template. Best path forward is to strip down a failing template until you find the root cause. Enabling logging in PS can help get more clarity as well (`Set-PnPTraceLog`), once an error occurs immediately run `Get-PnPException` and provide the details.

**Comment by jansenbe** [comment]
closing per no recent response

**Comment by arnieraju** [comment]
@jansenbe Thanks for the followup.  And apologies for not getting back to you here earlier.  

Unfortunately Set-PnPTraceLog and Get-PnPException did not offer much more to go off.  I did however end up finding the problem.  I tore the XML apart and through a process of elimination (and comparing the xml to a working template) found that the Language value was causing the problem.

The language was set to 3081 (English - Australia).  I changed this to 1033 and the template worked.   

Hopefully this helps people avoid the rabbit hole I ended up heading down. :)
